samza-core/src/main/java/org/apache/samza/operators/triggers/AnyTriggerImpl.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.operators.triggers;
+
+import org.apache.samza.operators.data.MessageEnvelope;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of an {@link AnyTrigger}
+ *
+ */
+public class AnyTriggerImpl<M extends MessageEnvelope> implements TriggerImpl<M> {
+
+  private final List<Trigger<M>> triggerList;
+
+  private final Map<TriggerImpl<M>, Boolean> triggerImpls = new HashMap<>();
+
+  public AnyTriggerImpl(AnyTrigger<M> anyTrigger) {
+    this.triggerList = anyTrigger.getTriggers();
+
+    for (Trigger<M> trigger : triggerList) {
+      triggerImpls.put(TriggerImpls.createTriggerImpl(trigger), false);
+    }
+  }
+
+  @Override
+  public void onMessage(M message, TriggerContext context, TriggerCallbackHandler handler) {
+    for (TriggerImpl<M> triggerImpl : triggerImpls.keySet()) {
```

Comment:
Good catch. In this case, the up-stream trigger called cancel on the `AnyTriggerImpl`. So, the rest of the triggers got invalidated. I have added the `cancel` in the callback handler to make it obvious.